src/SFA.DAS.ApprenticeFeedback.Jobs/Helpers/FeedbackTargetVariants/FeedbackTargetVariantBatchProcessor.cs

```diff
@@ -24,18 +25,17 @@ public FeedbackTargetVariantBatchProcessor(ILogger<FeedbackTargetVariantBatchPro
             _apprenticeFeedbackApi = apprenticeFeedbackApi;
         }
 
-        public async Task ProcessBatch(IEnumerable<FeedbackVariant> feedbackVariants, int batchSize)
+        public async Task ProcessBatch(List<FeedbackVariant> feedbackVariants, int batchSize)
         {
-            var variantsList = feedbackVariants.ToList();
+            var feedbackVariantBatches = feedbackVariants.ChunkBy(batchSize);
             bool isFirstBatch = true;
 
-            for (int i = 0; i < variantsList.Count; i += batchSize)
+            foreach (var batch in feedbackVariantBatches)
             {
-                var batch = variantsList.Skip(i).Take(batchSize).ToList();
                 bool clearStaging = isFirstBatch;
                 isFirstBatch = false;
 
-                bool isLastBatch = i + batchSize >= variantsList.Count;
+                bool isLastBatch = batch == feedbackVariantBatches.Last();
```

Comment:
That is interesting - in which case why would you not use .First() then instead of the bool?

Reply:
changed to use .first() and .last() to identify when to clear and merge staging